third_party/docfx-doclet-143274/src/main/java/com/microsoft/lookup/BaseLookup.java

```diff
@@ -313,4 +327,110 @@ private String getSeeTagRef(SeeTree seeTree) {
+      return "internal";
+    }
+    if (annotationNames.stream().anyMatch("Deprecated"::equals)
+        || hasDeprecatedJavadocTag(element)) {
+      return "deprecated";
+    }
+    if (annotationNames.stream().anyMatch("ObsoleteApi"::equals)) {
+      return "obsolete";
+    }
+    if (annotationNames.stream().anyMatch("BetaApi"::equals)) {
+      return "beta";
+    }
+    return null;
+  }
+
+  public String getStatusComment(T element) {
+    Map<String, Optional<String>> annotationComments = getAnnotationComments(element);
+
+    // Deprecated comments are determined by the Javadoc @deprecated block tag.
+    // See this#replaceBlockTags
+
+    List<String> comments = new ArrayList<>();
+    if (annotationComments.containsKey("InternalApi")) {
+      comments.add(createInternalOnlyNotice(annotationComments.get("InternalApi")));
+    }
+    if (annotationComments.containsKey("InternalExtensionOnly")) {
+      comments.add(
+          createInternalExtensionOnlyNotice(annotationComments.get("InternalExtensionOnly")));
+    }
+    if (annotationComments.containsKey("ObsoleteApi")) {
+      comments.add(createObsoleteNotice(annotationComments.get("ObsoleteApi")));
+    }
+    if (annotationComments.containsKey("BetaApi")) {
+      comments.add(createBetaNotice(annotationComments.get("BetaApi")));
+    }
+
+    if (comments.isEmpty()) {
+      return null;
+    }
+    return String.join("\n\n", comments);
+  }
+
+  private String createBetaNotice(Optional<String> customComment) {
+    return "<aside class=\"beta\">\n"
+        + "<p><strong>Beta</strong></p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "<p>This feature is covered by the <a href=\"/terms/service-terms#1\">Pre-GA Offerings "
+        + "Terms</a> of the Terms of Service. Pre-GA libraries might have limited support, and "
+        + "changes to pre-GA libraries might not be compatible with other pre-GA versions. For "
+        + "more information, see the launch stage descriptions.</p>\n"
+        + "</aside>\n";
+  }
+
+  private String createObsoleteNotice(Optional<String> customComment) {
+    return "<aside class=\"deprecated\">\n"
+        + "<p><strong>Obsolete</strong></p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "<p>This feature is stable for usage in this major version, but may be deprecated in a "
+        + "future release.</p>\n"
+        + "</aside>\n";
+  }
+
+  private String createInternalExtensionOnlyNotice(Optional<String> customComment) {
+    return "<aside class=\"special\">\n"
+        + "<p><strong>Internal Extension Only</strong>: This feature is stable for usage, but is "
+        + "not intended for extension or implementation.</p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "</aside>\n";
+  }
+
+  private String createInternalOnlyNotice(Optional<String> customComment) {

```

Comment:
now that I'm thinking about it, is `InternalOnly` meant to mean like internal Google-only? If so, theoretically that should be captured upstream in the protos via visibility labels such that we wouldn't even generate a client library containing anything that's internal only.

Reply:
Some methods are for internal only use, but are 'public' for technical reasons.

This is not "Google-only" in the sense that it contains server-side-only information. It's "internal only" in the sense that it is for use by the owning jar file only and is not to be used by applications.

See also https://github.com/googleapis/sdk-platform-java/blob/67a2c947fe6ee4bb3478b873a92d5e7c66eaa3b7/api-common-java/src/main/java/com/google/api/core/InternalApi.java#L38-L61